tests/_test_mscolab/test_files_api.py

```diff
@@ -24,49 +24,25 @@
     See the License for the specific language governing permissions and
     limitations under the License.
 """
-from flask_testing import TestCase
-import os
 import fs
 import pytest
 
-from mslib.mscolab.conf import mscolab_settings
 from mslib.mscolab.models import Operation
-from mslib.mscolab.server import APP
-from mslib.mscolab.file_manager import FileManager
 from mslib.mscolab.seed import add_user, get_user
-from mslib.mscolab.mscolab import handle_db_reset
 
 
-@pytest.mark.skipif(os.name == "nt",
```

Comment:
I check the tests on the windows tests system next day. I am not sure if the skip is needed or not.

Reply:
At least the stated reason for those skips became obsolete (start_method "fork" is no longer used), that's why I removed the skips. We should consider making use of GitHub's windows and macos runners to runner the test suite there automatically.

Reply:
yes, I used them also in the https://github.com/Open-MSS/mss-install too.

Reply:
test_files_api on windows is a lot better
```
(mssdev) C:\Users\r.bauer\PycharmProjects\2024\matrss\MSS>pytest -k test_files_api
========================================================= test session starts =========================================================
platform win32 -- Python 3.10.8, pytest-7.4.4, pluggy-1.3.0
rootdir: C:\Users\r.bauer\PycharmProjects\2024\matrss\MSS
configfile: pytest.ini
plugins: anyio-4.2.0, cov-4.1.0, reverse-1.7.0, timeout-2.2.0, xdist-3.5.0
timeout: 60.0s
timeout method: thread
timeout func_only: False
collected 541 items / 526 deselected / 1 skipped / 15 selected

tests/_test_mscolab/test_files_api.py::Test_Files::test_create_operation PASSED                                                  [  6%]
tests/_test_mscolab/test_files_api.py::Test_Files::test_list_operations ERROR                                                    [ 13%]
tests/_test_mscolab/test_files_api.py::Test_Files::test_get_operation_details PASSED                                             [ 20%]
tests/_test_mscolab/test_files_api.py::Test_Files::test_get_authorized_users PASSED                                              [ 26%]
tests/_test_mscolab/test_files_api.py::Test_Files::test_fetch_users_without_permission PASSED                                    [ 33%]
tests/_test_mscolab/test_files_api.py::Test_Files::test_fetch_users_with_permission PASSED                                       [ 40%]
tests/_test_mscolab/test_files_api.py::Test_Files::test_add_bulk_permissions ERROR                                               [ 46%]
tests/_test_mscolab/test_files_api.py::Test_Files::test_modify_bulk_permissions PASSED                                           [ 53%]
tests/_test_mscolab/test_files_api.py::Test_Files::test_delete_bulk_permissions PASSED                                           [ 60%]
tests/_test_mscolab/test_files_api.py::Test_Files::test_import_permissions ERROR                                                 [ 66%]
tests/_test_mscolab/test_files_api.py::Test_Files::test_update_operation FAILED                                                  [ 73%]
tests/_test_mscolab/test_files_api.py::Test_Files::test_delete_operation FAILED                                                  [ 80%]
tests/_test_mscolab/test_files_api.py::Test_Files::test_get_all_changes PASSED                                                   [ 86%]
tests/_test_mscolab/test_files_api.py::Test_Files::test_get_change_content PASSED                                                [ 93%]
tests/_test_mscolab/test_files_api.py::Test_Files::test_set_version_name ERROR                                                   [100%]
```
The errors all the same, like.

```
______________________________________ ERROR at setup of Test_MscolabVersionHistory.test_changes ______________________________________

mscolab_session_managers = (<Flask 'mslib.mscolab.app'>, <flask_socketio.SocketIO object at 0x000001FE6A2BDAB0>, <mslib.mscolab.chat_manager.ChatManager object at 0x000001FE6C24CD30>, <mslib.mscolab.file_manager.FileManager object at 0x000001FE6C24E740>)

    @pytest.fixture(scope="session")
    def mscolab_session_server(mscolab_session_managers):
        app, _, _, _ = mscolab_session_managers
        scheme = "http"
        host = "127.0.0.1"
        server = werkzeug.serving.make_server(host=host, port=0, app=app, threaded=True)
        port = server.socket.getsockname()[1]
        process = multiprocessing.Process(target=server.serve_forever, daemon=True)
>       process.start()

tests\fixtures.py:114:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
..\..\..\..\miniforge\envs\mssdev\lib\multiprocessing\process.py:121: in start
    self._popen = self._Popen(self)
..\..\..\..\miniforge\envs\mssdev\lib\multiprocessing\context.py:224: in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
..\..\..\..\miniforge\envs\mssdev\lib\multiprocessing\context.py:336: in _Popen
    return Popen(process_obj)
..\..\..\..\miniforge\envs\mssdev\lib\multiprocessing\popen_spawn_win32.py:93: in __init__
    reduction.dump(process_obj, to_child)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

obj = <Process name='Process-1' parent=13184 initial daemon>, file = <_io.BufferedWriter name=17>, protocol = None

    def dump(obj, file, protocol=None):
        '''Replacement for pickle.dump() using ForkingPickler.'''
>       ForkingPickler(file, protocol).dump(obj)
E       _pickle.PicklingError: Can't pickle <function db at 0x000001FE6A1F8C10>: it's not the same object as flask_migrate.cli.db

..\..\..\..\miniforge\envs\mssdev\lib\multiprocessing\reduction.py:60: PicklingError
_________________________________ ERROR at setup of Test_MscolabVersionHistory.test_set_version_name __________________________________

```

Reply:
I think we should first merge and afterwards look on other OS. Here already an amount of good changes.

Reply:
last time I looked in those docs, spawn is default on MacOS and Windows. And MacOS can also use fork but not Windows.  It is set explicity because of MacOS, there it overwrites the default.

Reply:
Yes, that is how it is described in the docs: https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods, with the caveat that it will change in 3.14.

I am not sure if the issue with forking multithreaded processes that is mentioned there could affect us.